### PR TITLE
BENCH: Masked array benchmarks

### PR DESCRIPTION
**Pull request by eric-wieser**
This adds some benchmarks that will hopefully detect the effects of #8910 and #8924.

Is this a reasonable set of things to benchmark, and is this the right way to structure them?

### CODE REVIEW

File: benchmarks/benchmarks/bench_ma.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+from __future__ import absolute_import, division, print_function
+
+from .common import Benchmark
+
+import numpy as np
+
+
+class MA(Benchmark):
```
Comment (eric-wieser) [inline benchmarks/benchmarks/bench_ma.py]: This was moved from `bench_core`. Is this a bad idea for `asv`?

Comment (charris) [inline benchmarks/benchmarks/bench_ma.py]: You mean, do we lose history?

Comment (charris) [inline benchmarks/benchmarks/bench_ma.py]: @pv Does this cause problems, or do cached results get recreated at some point.

**Comment by pv** [comment]
There's no cron job re-running for past commits.
But maybe there's nothing interesting in the history so you don't care.

**Comment by pv** [comment]
Of course, anyone interested can always re-run the benchmarks on their 
own machine(s).

**Comment by eric-wieser** [comment]
Would doing "from bench_core import _MA as MA" solve this problem?

**Comment by eric-wieser** [comment]
Sorry, I mean the other way around - move the source as I've done here, but leave the test accessible under `bench_core.MA`

**Comment by pv** [comment]
I suggest to not add any workarounds and just ignore the issue. Past 
benchmark results can always be regenerated if there's some need for them.

**Comment by charris** [comment]
> I suggest to not add any workarounds ...

Agree. The reorganization looks natural going forward.

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**